src/modules/modals/ImportPreview.vue

```diff
@@ -0,0 +1,299 @@
+<template>
+	<div>
+		<table class="file_import__preview">
+			<caption>
+				{{ t('tables', 'Preview') }}
+			</caption>
+			<thead>
+				<tr>
+					<th>{{ t('tables', 'Column') }}</th>
+					<th>{{ t('tables', 'Data') }}</th>
+				</tr>
+			</thead>
+			<tbody>
+				<tr v-for="(column, colIndex) in columnsConfig" :key="colIndex">
+					<td>
+						<div class="row space-L space-T space-B">
+							<div class="column-title">
+								{{ column.title }}
+							</div>
+							<div class="row no-padding-on-mobile">
+								<div class="col-4 action-selections space-B space-T">
+									<NcCheckboxRadioSwitch v-if="createMissingColumns" :checked.sync="columnsConfig[colIndex].action" value="new" name="columnActionSelection" type="radio" @update:checked="$forceUpdate()">
+										{{ t('tables', 'New column') }}
+									</NcCheckboxRadioSwitch>
+									<NcCheckboxRadioSwitch :checked.sync="columnsConfig[colIndex].action" value="exist" name="columnActionSelection" type="radio" @update:checked="$forceUpdate()">
+										{{ t('tables', 'Exist column') }}
+									</NcCheckboxRadioSwitch>
+									<NcCheckboxRadioSwitch v-if="!createMissingColumns" :checked.sync="columnsConfig[colIndex].action" value="ignore" name="columnActionSelection" type="radio" @update:checked="$forceUpdate()">
+										{{ t('tables', 'Ignore') }}
+									</NcCheckboxRadioSwitch>
+								</div>
+							</div>
+
+							<div v-if="column.action === 'new'" class="row no-padding-on-mobile">
+								<NcButton @click="editColumn(colIndex)">
+									{{ columnsConfig[colIndex].typeLabel }}
+									<template #icon>
+										<TextLongIcon v-if="['text', 'text-line', 'text-rich'].includes(column.typeFull)" />
+										<LinkIcon v-if="column.typeFull === 'text-link'" />
+										<CounterIcon v-if="column.typeFull === 'number'" />
+										<StarIcon v-if="column.typeFull === 'number-stars'" />
+										<ProgressIcon v-if="column.typeFull === 'number-progress'" />
+										<SelectionIcon v-if="column.type === 'selection'" />
+										<DatetimeIcon v-if="column.type === 'datetime'" />
+									</template>
+								</NcButton>
+							</div>
+
+							<div v-if="columnsConfig[colIndex].action === 'exist'" class="row no-padding-on-mobile">
+								<NcSelect v-model="columnsConfig[colIndex].existColumn" :options="existColumnOptions" :aria-label-combobox="t('tables', 'Exist column')" />
```

Comment:
> Maybe it's better to hide existing column options where there is a datatype mismatch

I think we should let users be free to select existing column since the detected data type might not be really exactly and the matching datatype with existing columns might have many cases.